protocols/jabber/iq.c

```diff
@@ -1006,12 +1006,23 @@ static xt_status jabber_iq_disco_server_response(struct im_connection *ic,
                                                  struct xt_node *node, struct xt_node *orig)
 {
 	struct jabber_data *jd = ic->proto_data;
-	struct xt_node *query, *id;
+	struct xt_node *query, *id, *c;
+	char *feature;
 
 	if (!(query = xt_find_node(node->children, "query"))) {
 		return XT_HANDLED;
 	}
 
+	c = query->children;
+
+	while ((c = xt_find_node(c, "feature"))) {
+		feature = xt_find_attr(c, "var");
+		if (feature) {
+			jd->features = g_slist_append(jd->features, g_strdup(feature));
+		}
+		c = c->next;
+	}
+
```

Comment:
Hmm so this receives a disco response and puts every item in jd->features. If a new response arrives, there's not really any way to forget about a feature. I'm not saying there /should/ be, but the other part of the code kinda wants to set things to NULL if it's gone, so there's an answer to that question.

On the other hand: if i'm reading this right, the only purpose of saving features in this linked list is to keep it around to later attach the handlers in block.c.

Why not just use `xt_find_node_by_attr` as with the other features, and attach the handlers to the prpl here?